services/std_svc/sdei/sdei_main.c

```diff
@@ -925,32 +924,43 @@ uint64_t sdei_smc_handler(uint32_t smc_fid,
 		ret = sdei_event_register(x1, x2, x3, x4, x5);
 		SDEI_LOG("< REG:%ld\n", ret);
 		SMC_RET1(handle, ret);
-		break;
 
 	case SDEI_EVENT_ENABLE:
 		SDEI_LOG("> ENABLE(n:%d)\n", (int) x1);
 		ret = sdei_event_enable(x1);
 		SDEI_LOG("< ENABLE:%ld\n", ret);
 		SMC_RET1(handle, ret);
-		break;
 
 	case SDEI_EVENT_DISABLE:
 		SDEI_LOG("> DISABLE(n:%d)\n", (int) x1);
 		ret = sdei_event_disable(x1);
 		SDEI_LOG("< DISABLE:%ld\n", ret);
 		SMC_RET1(handle, ret);
-		break;
 
 	case SDEI_EVENT_CONTEXT:
 		SDEI_LOG("> CTX(p:%d):%lx\n", (int) x1, read_mpidr_el1());
 		ret = sdei_event_context(handle, x1);
 		SDEI_LOG("< CTX:%ld\n", ret);
 		SMC_RET1(handle, ret);
-		break;
 
 	case SDEI_EVENT_COMPLETE_AND_RESUME:
 		resume = 1;
-		/* Fall through */
+		SDEI_LOG("> COMPLETE(r:%d sta/ep:%lx):%lx\n", resume, x1,
```

Comment:
I think that we should either live with the MISRA fail (and mark as ignore), or move this into a separate (static inline) function that can be called from both the SDEI_EVENT_COMPLETE_AND_RESUME and SDEI_EVENT_COMPLETE case statements.

Reply:
The SMC_RETs make this difficult to do in a static inline function. Personally I don't see a problem with this change as it gives the same form to each switch clause - rather than having one case of fall-through.

However, I'll remove this change.